src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -698,7 +698,13 @@ InterpreterCreateQuery::TableProperties InterpreterCreateQuery::getTableProperti
     }
     else if (create.select)
     {
-        Block as_select_sample = InterpreterSelectWithUnionQuery::getSampleBlock(create.select->clone(), getContext());
+        Block as_select_sample;
+        if (create.is_window_view)
+            as_select_sample = InterpreterSelectWithUnionQuery(
+                                   create.select->clone(), getContext(), SelectQueryOptions(QueryProcessingStage::FetchColumns).analyze())
+                                   .getSampleBlock();
```

Comment:
I would rather move this into `InterpreterSelectWithUnionQuery::getSampleBlock` and add argument into this method for stage enum for select query options.

Reply:
Yes, select query options has already moved into `InterpreterSelectWithUnionQuery::getSampleBlock`